pkg/pool-utils/test/BasePoolController.test.ts

```diff
@@ -0,0 +1,236 @@
+
+      expect(await poolController.getMetadata()).to.equal(METADATA);
+    });
+
+    it('non-owner cannot set metadata', async () => {
+      await expect(poolController.connect(other).updateMetadata(METADATA)).to.be.revertedWith('CALLER_IS_NOT_OWNER');
+    });
+  });
+
+  context('pool controller is initialized', () => {
+    sharedBeforeEach('initialize pool controller', async () => {
+      await poolController.bindPool(pool.address);
+    });
+
+    it('owner can set the swap fee', async () => {
+      await poolController.connect(admin).setSwapFeePercentage(NEW_SWAP_FEE);
+
+      expect(await pool.getSwapFeePercentage()).to.equal(NEW_SWAP_FEE);
+    });
+
+    it('enforces owner permissions', async () => {
+      await expect(poolController.connect(other).setSwapFeePercentage(NEW_SWAP_FEE)).to.be.revertedWith(
+        'CALLER_IS_NOT_OWNER'
+      );
+    });
+
+    it('can be transferred', async () => {
+      await poolController.transferOwnership(other.address);
+
+      await expect(poolController.connect(admin).setSwapFeePercentage(NEXT_SWAP_FEE)).to.be.revertedWith(
+        'CALLER_IS_NOT_OWNER'
+      );
+
+      await poolController.connect(other).setSwapFeePercentage(NEXT_SWAP_FEE);
+
+      expect(await pool.getSwapFeePercentage()).to.equal(NEXT_SWAP_FEE);
+    });
+
+    it('can renounce ownership', async () => {
+      await poolController.renounceOwnership();
+
+      await expect(poolController.connect(admin).setSwapFeePercentage(NEXT_SWAP_FEE)).to.be.revertedWith(
+        'CALLER_IS_NOT_OWNER'
+      );
+
+      expect(await poolController.owner()).to.equal(ZERO_ADDRESS);
+    });
+
+    it('cannot be paused by owner', async () => {
+      await expect(pool.setPaused(true)).to.be.revertedWith('SENDER_NOT_ALLOWED');
+    });
+
+    context('pause authorized', async () => {
+      sharedBeforeEach('authorize governance to pause', async () => {
+        const action = await actionId(pool.instance, 'setPaused');
+        await vault.grantRole(action, governance.address);
+      });
+
+      it('still cannot be paused by owner', async () => {
+        await expect(pool.setPaused(true)).to.be.revertedWith('SENDER_NOT_ALLOWED');
+      });
+
+      it('can be paused by governance', async () => {
+        await pool.instance.connect(governance).setPaused(true);
+
+        expect(await pool.isPaused()).to.be.true;
+      });
+    });
+
+    describe('Asset Managed pool', () => {
+      let assetManagerContract: Contract;
+      let assetManagers: string[];
+      let assetManagedPool: WeightedPool;
+      let poolId: string;
+
+      const poolConfig = {
+        targetPercentage: 3,
+        upperCriticalPercentage: 4,
+        lowerCriticalPercentage: 2,
+      };
+
+      sharedBeforeEach('deploy asset managed pool', async () => {
+        const lendingPool = await deploy('v2-asset-manager-utils/MockAaveLendingPool', { args: [] });
+        const aaveRewardsController = await deploy('v2-asset-manager-utils/MockAaveRewards');
```

Comment:
This is way overkill: there is no need to introduce dependencies with the mocks for Aave asset managers here.

We should simply use MockAssetManager (which admittedly is currently in a very poor state and needs some improvements, for example emitting an event in `setConfig` that we can look for in the tests here).